### ISSUE TITLE
check lang detect, which seems to be setting lang = NULL for some obviously english cases

### ISSUE THREAD

**Issue by fhamborg** [fhamborg/news-please#128]
https://mail.google.com/mail/u/0/#inbox/FMfcgxwDrvDbsjsXSDHBlJVfKfCvGfMQ

**Comment by fhamborg** [comment]
```
Washington (CNN) Senate Republicans voted to advance to floor debate on their efforts to repeal and replace Obamacare.
Vice President Mike Pence cast the tiebreaking vote.
In a dramatic moment, Sen. John McCain returned from Arizona to applause from fellow senators. He cast a necessary Republican vote for the motion after two GOP senators -- Susan Collins and Lisa Murkowski -- sided with all Democrats in opposition.
As the vote began, protesters in the Senate gallery shouted "kill the bill" and "shame, shame, shame!"
The vote came as Senate Majority Leader Mitch McConnell and President Donald Trump dared their fellow Republicans to block their seven-year campaign promise to repeal and replace Obamacare.
The next step is floor debate on the legislation to overhaul the Affordable Care Act even though there aren't any guarantees the votes are there to eventually pass it -- and it's unclear what a final bill will look like.
The vote was up in the air until the last moments, when several Republican holdouts announced their support, including Sens. Rand Paul, Dean Heller, Rob Portman and Shelley Moore Capito.
Trump, who has repeatedly said he's ready to sign any repeal legislation, celebrated the vote, which creates a path to give him the major congressional victory that's eluded the White House thus far.
"I'm very happy to announce that with zero of the Democrats' votes, the motion to proceed on health care has moved past and now we move forward toward truly great health care for the American people. We look forward to that. This was a big step," Trump said at a White House news conference.
"I want to thank Sen. John McCain," he added. "A very brave man. He made a tough trip to get here."
But while Trump was happy with the result, he also acknowledged there's a long road ahead for the legislation and criticized the two GOP senators who opposed the motion.
"We had two Republicans that went against us, which is very sad, I think," Trump said. "It's very, very sad -- for them. But I'm very, very happy with the result. I believe now we will, over the next week or two, come up with a plan that's going to be really, really wonderful for the American people."
Democrats are united against the bill, saying it would end health care coverage for millions of Americans.
"Anyone who thinks this is over is sadly mistaken," said Minority Leader Chuck Schumer, D-New York. "There are many, many Republicans who don't like this bill, who don't want to vote for it. They are under enormous pressure to vote for it. ... Anyone who voted to move to proceed and certainly anyone who votes to send this bill to conference is virtually, certainly, voting to kick millions off health care, to make it much harder to get coverage if you have a pre-existing condition, to slash Medicaid and give a huge amount of tax cuts to the rich."
Sen. Bernie Sanders Monday called the bill the "cruelest, most destructive and irresponsible piece of legislation ever brought to the United States Senate in the modern history of this country."
In a speech at the NAACP national convention, highlighted the possible effects of the bill's provisions, which include cutting Medicaid, defunding Planned Parenthood and roadblocks for those with pre-existing conditions.
JUST WATCHED McCain returns to Senate floor (full speech) Replay More Videos ... MUST WATCH McCain returns to Senate floor (full speech) 15:25
Paul, Heller back motion
McConnell won over key holdouts on the procedural vote, but those senators haven't said they will back the final measure.
Paul, a Kentucky Republican, said he would support the procedural motion to open debate on the health care bill, so long as leadership guarantees a vote on a full repeal of Obamacare.
"If this is indeed the plan, I will vote to proceed and I will vote for any all measures that are clean repeal," Paul tweeted. Such an amendment would be expected to fail, however.
Heller, who has complained that Obamacare repeal efforts could hurt Nevada residents dependent on Medicaid, nevertheless said he'd vote to move forward. Heller is facing a tough re-election campaign in Nevada next year.
"Obamacare isn't the answer, but doing nothing to try to solve the problems it has created isn't the answer either," Heller said in a statement. "If the final product isn't improved for the state of Nevada, then I will not vote for it; if it is improved, I will support it."
Republican leaders Tuesday worked thread any needle they can to appease other holdouts.
"The only goal is to get onto the bill. Nothing happens until we do, so that's the only goal," a Republican aide said.
"These are the moments legislatively when you get creative. We're getting creative."
JUST WATCHED Trump: Senate GOP can end Obamacare nightmare Replay More Videos ... MUST WATCH Trump: Senate GOP can end Obamacare nightmare 02:28
Now what?
Leadership was floating a strategy Tuesday morning that lays out a series of proposals that attempt to give everyone something of what they want, even though nearly every element is destined for failure on the floor during the amendment process.
That strategy is as follows, in terms of amendment order:
The full repeal bill that Paul backs.
A bill including the "consumer freedom amendment" that Sens. Ted Cruz and Mike Lee originally advocated for, which includes a request from Sen. Rob Portman for $100 billion in additional Medicaid funding. That would require 60 votes to pass.
The GOP bill proposed earlier this month that McConnell pulled when too many Republicans opposed it.
Finally, a "skinnier" repeal bill that repeals Obamacare's individual and employer mandates.
There are also several parts of any legislation that could end up stripped from the final bill thanks to Senate rules that will require 60 votes for passage rather than 50.
The Senate parliamentarian says a provision that would allow insurers to charge higher premiums to older enrollees, as compared to younger ones, would be subject to the 60-vote requirement. Obamacare limited insurers to charging those in their 50s and 60s up to three times the rates of younger consumers. The Senate bill would widen that to five times, hiking the premiums that older enrollees must pay.
This provision has drawn sharp criticism from the influential AARP, which has slammed it as an "age tax."
The parliamentarian is also reviewing whether to allow several key changes Republicans want to make to Obamacare's insurance regulations. Obamacare mandated that insurers cover 10 essential health benefits, including maternity, substance abuse and prescription drugs. The Senate bill would allow states to apply for a waiver from this requirement so insurers could offer skimpier polices with lower premiums.
Conservatives will not be pleased if this waiver section is removed.
This story has been updated.
```

**Comment by lukasgebhard** [comment]
To help with debugging, here is a list of 62K row IDs where `lang` is `NULL` but langdetect outputs `'en'`
[lang_null_langdetected_en.zip](https://github.com/fhamborg/news-please/files/3831829/lang_null_langdetected_en.zip)

**Comment by fhamborg** [comment]
The issue is due to logical problem in https://github.com/fhamborg/news-please/blob/3b93656a87305a2c896d8d5bba1acbca6144f21d/newsplease/pipeline/extractor/comparer/comparer_Language.py#L43
Trying some things to fix

**Comment by fhamborg** [comment]
fixed by 23cffaf5df407cddcd0f941dec87a4619cf6f476
(not available as pypi yet)
